File: server/php/src/Application/Domain/Payments/PaymentDetails/ExternalTokenLinked.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class ExternalTokenLinked
{
    public ?string $computedToken = null;
    public ?string $gtsComputedToken = null;
    public ?string $generatedToken = null;

    public function toArray(): array
    {
        return [
            'computedToken' => $this->computedToken,
            'gtsComputedToken' => $this->gtsComputedToken,
            'generatedToken' => $this->generatedToken,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/external_token_linked.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class ExternalTokenLinked
          attr_accessor :computed_token, :gts_computed_token, :generated_token

          def initialize(computed_token: nil, gts_computed_token: nil, generated_token: nil)
            @computed_token = computed_token
            @gts_computed_token = gts_computed_token
            @generated_token = generated_token
          end

          def to_h
            {
              'computedToken' => @computed_token,
              'gtsComputedToken' => @gts_computed_token,
              'generatedToken' => @generated_token
            }
          end
        end
      end
    end
  end
end
